### PR TITLE
[Security] Dispatch an event when "logout user on change" steps in

### PR DESCRIPTION
**Pull request by Simperfit**
This documents the new event implemented in https://github.com/symfony/symfony/pull/31138

### CODE REVIEW

File: components/security/authentication.rst [hunk 1]
```diff
@@ -314,6 +315,9 @@ order to give your user a welcome flash message every time they log in.
 The ``security.switch_user`` event is triggered every time you activate
 the ``switch_user`` firewall listener.
 
+The ``security.logout_on_change`` event is triggered when a token has been deauthenticated
+because of a user change, it can help you doing some clean-up task when a logout has been triggered.
```
Comment (OskarStark) [inline components/security/authentication.rst]: Please add a versionadded directive here about the new event

Comment (Simperfit) [reply components/security/authentication.rst]: done

**Review by OskarStark — APPROVED** [review]

File: components/security/authentication.rst [hunk 2]
```diff
@@ -283,6 +283,7 @@ security.authentication.success  ``AuthenticationEvents::AUTHENTICATION_SUCCESS`
 security.authentication.failure  ``AuthenticationEvents::AUTHENTICATION_FAILURE``  :class:`Symfony\\Component\\Security\\Core\\Event\\AuthenticationFailureEvent`
 security.interactive_login       ``SecurityEvents::INTERACTIVE_LOGIN``             :class:`Symfony\\Component\\Security\\Http\\Event\\InteractiveLoginEvent`
 security.switch_user             ``SecurityEvents::SWITCH_USER``                   :class:`Symfony\\Component\\Security\\Http\\Event\\SwitchUserEvent`
+security.logout_on_change        ``SecurityEvents::LOGOUT_ON_CHANGE``              :class:`Symfony\\Component\\Security\\Http\\Event\\LogoutOnChangeEvent`
```
Comment (chalasr) [inline components/security/authentication.rst]: the new event uses the Event class FQCN as name

**Review by chalasr — CHANGES_REQUESTED** [review]

**Comment by chalasr** [comment]
The code PR has been merged

**Comment by OskarStark** [comment]
Thank you Hamza.

**Pull request merged**